demo/parser_demo.cpp

```diff
@@ -129,14 +129,61 @@ class PrintVisitor : public ast::Visitor {
             depth--;
         }
     };
-    void visit(ast::Expression* node) override{
-
+    void visit(ast::Expression* node) override {
+        if (node == nullptr) {
+            fmt::print("{:|>{}}- [Expression]> null\n", "", depth);
+        } else {
+            // if binary
+            if (node->operand2) {
```

Comment:
Explicit is always better. This is not JavaScript 😜😜
```suggestion
            if (node->operand2 != nullptr) {
```

Reply:
Actually, this entire `if` statement can simply wrap just the call to `node->operand2->accept`